### ISSUE TITLE
S-Prompts Learning with Pre-trained Transformers: An Occam’s Razor for Domain Incremental Learning

### ISSUE THREAD

**Issue by HYUNMIN-HWANG** [One-Day-One-Paper/One-Day-One-Paper#301]
**Domain Incremental Learning (DIL)** : learning the prompts independently across domains with pre-trained transformers → the prompting can achieve the best for each domain.

동기 : 그동안 continual learning들은 buffer를 사용하거나 rehearsal, distillation이 필요했는데, 해당 논문에서는 따른 buffer 필요 없이 오로지 prompt만 사용해서 continual learning을 할 것이다. (exemplar-free DIL)

![Untitled](https://user-images.githubusercontent.com/70581043/194079257-b28cfe36-9218-4689-8ddf-7254338df816.png)

## S-prompts

특징 요약

- learning the prompts independently across domains so that the prompting can achieve the best for each domain.
- frozen pre-trained transformer to extract general features
- independent prompting reaches the best fit on each domain.
- cross-entropy 사용해서 classification
- tunes the current domain related prompts
- classifier with all the rest neural network components

- domain : $s$
- input $\mathbb{D}_s = \{x_i^s,y_i^s\}_{i=1}^{N_s}$
    
    $x_i^s$ : $i$-th input image from domain $s$
    
    $y_i^s$ : label
    
- pre-trained transformers (ViT) : to extract highly expressive low-level features
- prompts : to encode the high-level features

![Untitled 1](https://user-images.githubusercontent.com/70581043/194079275-04dd299d-9300-4d25-973c-a8a72c478a62.png)
(1) Image S-Prompts 

(S-iPrompts)

$P_x^i\in \mathbb{R}^{L_i \times D_i}$ : continuous learnable parameters, input to the pre-trained ViT

$x=[x_{img}, P_s^i, x_{cls}]$ : whole embedding

새로운 domain (s+1)이 들어오면 $P_{s+1}^i$를 추가한다.

classification

- FC layer 거친 후 → softmax
    
    $\hat{y} = \sigma (W_sf(x)+b_s)$
    

(2) Language-Image S-Prompts (S-liPrompts)

pre-trained CLIP 사용

CLIP에서 pair 한 이미지와 텍스트를 사용한 것처럼 pair of an image-end prompt and a language-end prompt domain by domain, with the pre-trained CLIP always being fixed

$P_s^l=\{v_1, v_2, ..., v_M\}\in\mathbb{R}^{L_l\times D_t}$ : learnable context prompt

$t_j=\{P_s^l, c_j\}$ : whole input of the text encoder ($c_j$ : [CLASS] token)

classification

- CLIP에서 사용하는 contrastive loss 사용
    
    $p(y_j|x) = \frac{exp(<f(x), g(t_j)>)}{\sum_{k=1}^C exp(<f(x), g(t_k)>)}$
    
- 그 이후, softmax activation
